### DISCUSSION TITLE
Restarting driver without rebooting

### DISCUSSION THREAD

**Discussion by vincegod** [Louisvdw/dbus-serialbattery#392] [Q&A] [answered]
What’s the easiest way to restart the driver after making changes without having to reboot?

I have seen it somewhere but can’t find it now :(

Thanks

Vince

**Comment by HotteX** [comment]
If you edited in /data/etc/ you first have to copy the file, otherwise it will be ignored until restart:
For utils.py that would be:
`cp -a /data/etc/dbus-serialbattery/utils.py /opt/victronenergy/dbus-serialbattery/
`
Then you need to restart the tool:
Change the device accordingly to yours.
```
/opt/victronenergy/serial-starter/stop-tty.sh /dev/ttyACM0
/opt/victronenergy/serial-starter/start-tty.sh /dev/ttyACM0
```

**Comment by TimGFoley** [comment]
Actually I see there is now a restartsvc shell file that runs the command:

svc -d -u /service/dbus-serialbattery

Seemed to work for me.

**Comment by Louisvdw** [comment, marked as answer]
Instead of making a manual copy I would suggest running the script to do that.

So first
> sh /data/etc/dbus-serialbattery/reinstalllocal.sh

Then either 
> /opt/victronenergy/serial-starter/stop-tty.sh /dev/ttyUSB0
/opt/victronenergy/serial-starter/start-tty.sh /dev/ttyUSB0

or (this one does not work for everyone but is the same as the svc command that Tim shared)
> /data/etc/dbus-serialbattery/restartservice.sh)

**Reply by vincegod** [reply]
Thanks for all the help and hopefully this thread will make it easier for others to find this information in future.
